server/log_rpc_server_test.go

```diff
@@ -553,16 +553,17 @@ func TestGetLatestSignedLogRoot2(t *testing.T) {
 			noCommit: true,
 		},
 		{
-			// Test error case where storage fails to commit the tx.
-			req:       getLogRootRequest1,
-			errStr:    "commit",
-			commitErr: errors.New("commit() error"),
+			// Test error case where the log root could not be read
+			req:      getLogRootRequest1,
+			errStr:   "rpc error: code = Internal desc = Could not read current log root: logRootBytes too short",
+			noCommit: true,
 		},
 		{
 			// Test normal case where a root is returned correctly.
 			req:         getLogRootRequest1,
 			wantRoot:    trillian.GetLatestSignedLogRootResponse{SignedLogRoot: signedRoot1},
 			storageRoot: *signedRoot1,
+			noCommit:    true,
```

Comment:
The success case *should* commit.  Were you trying to do something else here?

Reply:
I fixed this issue by committing the TX before returning the root in the case of req.FirstTreeSize = 0. (See the change in log_rpc_server.go)